### ISSUE TITLE
train title and abstract only results in poor abstract extraction

### ISSUE THREAD

**Issue by de-code** [kermitt2/grobid#430]
Hi Patrice,

I am re-training the header model currently with the title and abstract only.
It does seem to work fine with the title. But the abstract extraction afterwards is very poor. It often returns authors for example. I've trained it with 100 and 1000 documents using GROBID 0.5.2 (will try 0.5.4 but it had abstract regressions). I've actually then ran the model using 0.5.3. (I will aim to keep the version consistent in the future)

My process looks roughly like:
- generate raw training data
- remove all tagging (except for `lb`)
- find title and abstract and add tags (auto-annotated tei)
- train model
- evaluate:
  - auto-annotated tei
  - grobid model on training pdf
  - grobid model on validation/test pdf

For the evaluation on auto-annotated tei I am getting a f1-score between 0.85 and 0.90 (levenshtein). Which shows that while not perfect, the training data is not that bad (some documents may not have header data).

Evaluation of the grobid model on the training pdf then ends up far below 0.10 and looking at samples it's not even close. The default model is much better.

The auto-annotated training TEI XML would look something like this:

```xml
<tei><teiHeader>
                <fileDesc xml:id="..."/>
        </teiHeader>
        <text><front><lb/>


        <docTitle><titlePart>Title line 1<lb/> and line 2<lb/></titlePart></docTitle>

        Author One and Author Two<lb/> 1<lb/>

        ABSTRACT<lb/> <div type="abstract">A long and informative<lb/> abstract...</div>.

        </front></text></tei>
```

Is that something that would get solved by using the Clusteror? Am I doing something wrong?

**Comment by de-code** [comment]
I've ran a few more tests.

I was doing the training with 0.5.3 throughout and 0.5.4. For 50, 100, 200 and 500 training samples (not 1000 yet because it takes so long). I am evaluating it on 100 validation samples which could be larger but are randomly selected (but fixed, i.e. it won't change per run or future runs).

In both cases it trained the title alright. Although still below the default model. Slightly better in 0.5.4.

The abstract training really doesn't seem to work in 0.5.3 as described in the issue above (with a f1 score of far below 0.1 - in fact it was best with just 50 training samples).

Training 0.5.4 was better with a f1 score below 0.5. There is no sign of it improving with more training samples. In fact using 500 training samples yielded the worst result. As with 0.5.3, the evaluation results on training samples themselves are not much different than the validation samples (in some instances even worse).

**Comment by de-code** [comment]
Hi @kermitt2 

Thank you again for getting back on that. I have implemented a change and with DeLFT it is more visible what the actual training data looks like.

The title extraction has actually improved.

The abstract however is another story and I am a bit confused. Basically looking at two author submitted manuscripts used for training. Unfortunately not public but I can try to find public ones later. In both cases the abstract is on the second page, with a clear "abstract" header.

Example 1:
In full text mode, the header is sent to DeLFT which annotates the abstract correctly. The TEI output however cuts the last word off (after the hyphen). If I select `Process Header Document` the text sent to DeLFT ends before the abstract.

Example 2:
GROBID fails to extract the abstract also in full text mode. The text sent to DeLFT ends before the abstract begins. Trying to debug GROBID I came across [Document.getHeaderFeatured](https://github.com/kermitt2/grobid/blob/4f3a906fb7fc9e4eec060ce64420d3af063986f5/grobid-core/src/main/java/org/grobid/core/document/Document.java#L653) and [Document.getHeader](https://github.com/kermitt2/grobid/blob/4f3a906fb7fc9e4eec060ce64420d3af063986f5/grobid-core/src/main/java/org/grobid/core/document/Document.java#L653). The latter pointing out `it is now covered by the CRF segmentation model`. The rules in `getHeader` are a bit hard to follow. And the result is actually thrown away, using `this.blockDocumentHeaders` instead. I am assuming the abstract is still meant to be part of the header? Why isn't it just using the CRF segmentation model? Is that something that would change with the `Clusteror`? (any idea when that will be implemented?)

I believe I used GROBID 0.5.4 (in non-DeLFT mode) to generate the raw TEI/header training data (where I then change the annotation based on the JATS XML). i.e. it seem to have segmented the data correctly for that purpose.

What would you suggest to improve the extraction?

**Comment by de-code** [comment]
Further debugging...

As for Example 1:
I could imagine that this is related to the [stop condition in getHeader](https://github.com/kermitt2/grobid/blob/4f3a906fb7fc9e4eec060ce64420d3af063986f5/grobid-core/src/main/java/org/grobid/core/document/Document.java#L1013):
```java
            if ((i > 6) && (i > (blocks.size() * 0.6))) {
                if (beginBody != -1) {
                    return res;
                } else
                    return null;
            }
```

That should probably be disabled if we are only processing the header. But then this method is probably not the way to go anyway.


For Example 2:
I guess it is just falling back to the first page because the heuristics can't find anything. The segmentation model seem to have correctly identified the end of the header, after the abstract.

**Comment by kermitt2** [comment]
You might prefer to use only the segmentation model approach to identify the header, and not the heuristics based approach. Heuristics are used for processHeader because it's a bit more accurate in average than using the segmentation model (clear lack of training data for this model). Heuristics approach is also used in priority for processFulltext, segmentation model as fallback.

When I say on average, it's based on PMC files, and it might not be the case for your documents. Heuristics are heuristics, usually if you touch something somewhere there, it breaks elsewhere :) This is really supposed to be removed, and I was expecting to remove that since a while, but addition of training data for the segmentation model did not happen so far. 

For generating training data for the segmentation model (and header model), only the segmentation model is used, and not the heuristics (which makes sense because we don't particularly care about accuracy here, simply speed-up the annotation process). It explains why the segmentation is different from the "normal" process which uses the heuristics first. 

See #476 on the same subject!

**Comment by kermitt2** [comment]
There's now no more heuristics used for finding the header (from version 0.6.1), the segmentation model is always used for zoning. All the training data for the header model is derived from the segmentation model too.

**Comment by de-code** [comment]
Okay, I think this can be closed or could have been closed a long time ago. I added a note as suggested.
